app/models/deck.rb

```diff
@@ -4,22 +4,49 @@ class Deck < ApplicationRecord
 
   def generate_countries_deck
     data = CountriesService.fetch_all
+    create_cards(data)
+  end
 
+  def create_cards(data)
     data.each do |row|
-      map = MapService.get_map({ lat: row[:latlng][0], lon: row[:latlng][1] }) if row[:latlng][0] && row[:latlng][1]
+      map = get_maps(row)
       map && cards.create!(
         name: row[:name],
         capital: row[:capital],
         population: row[:population],
         lat: row[:latlng][0],
         lon: row[:latlng][1],
-        timezones: row[:timezones][0],
+        timezones: current_time(row[:timezones][0]),
         currencies: row[:currencies][0][:name],
-        languages: row[:languages].map { |lang| lang[:name] },
+        languages: languages(row[:languages]),
         flag: row[:flag],
         map: map
       )
     end
   end
+
```

Comment:
Created methods to abstract the logic out of the object values to help with readability and to dry up the code